File: backend/Models/Ticket.cs
using System;
using System.Linq;
using Backend.Constants;

namespace Backend.Models;

public class Ticket
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Name { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? Summary { get; set; }
    public string? ImageUrl { get; set; }
    public string Status { get; set; } = ApplicationConstants.TicketStatus.New;
    public string Resolution { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public static string[] AllowedStatuses => ApplicationConstants.TicketStatus.AllStatuses;

    public bool IsValidStatus(string status)
    {
        return AllowedStatuses.Contains(status, StringComparer.OrdinalIgnoreCase);
    }
}

File: frontend/src/types/Ticket.ts
export interface Ticket {
  id: string;
  name: string;
  email: string;
  description: string;
  summary?: string;
  imageUrl?: string;
  status: string;
  resolution: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTicketRequest {
  name: string;
  email: string;
  description: string;
}

export interface UpdateTicketRequest {
  status?: string;
  resolution?: string;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface TicketFilters {
  status?: string;
  search?: string;
  page?: number;
  pageSize?: number;
}
